File: gradle-plugins/react/brownfield/src/main/kotlin/com/callstack/react/brownfield/shared/Logging.kt
package com.callstack.react.brownfield.shared

import com.callstack.react.brownfield.shared.Constants.PLUGIN_NAME

object Logging : BaseProject() {
    fun error(
        message: Any,
        error: Throwable,
    ) {
        project.logger.error(getFormattedMessage(message), error)
    }

    fun warn(
        message: Any,
    ) {
        project.logger.warn(getFormattedMessage(message))
    }

    fun info(message: Any) {
        project.logger.info(getFormattedMessage(message))
    }

    fun log(message: Any) {
        println(getFormattedMessage(message))
    }

    private fun getFormattedMessage(message: Any): String {
        return " $PLUGIN_NAME :: $message"
    }
}

File: packages/react-native-brownfield/src/expo-config-plugin/logging.ts
const LOG_TAG = '[react-native-brownfield]';

export class Logger {
  private static debug: boolean = false;

  static setIsDebug(enabled: boolean) {
    this.debug = enabled;
  }

  static logInfo(message: string, ...args: any[]) {
    console.log(`${LOG_TAG} ${message}`, ...args);
  }

  static logDebug(message: string, ...args: any[]) {
    if (!this.debug) {
      return;
    }

    console.debug(`${LOG_TAG} ${message}`, ...args);
  }

  static logWarning(message: string, ...args: any[]) {
    console.warn(`${LOG_TAG} ${message}`, ...args);
  }

  static logError(message: string, ...args: any[]) {
    console.error(`${LOG_TAG} ${message}`, ...args);
  }
}
